versal2.mk

```diff
@@ -0,0 +1,167 @@
+# OP-TEE
+#################################################################################
+
+OPTEE_OS_COMMON_FLAGS += CFG_TEE_CORE_LOG_LEVEL=2 CFG_TEE_TA_LOG_LEVEL=2
+
+optee-os: optee-os-common
+
+optee-os-clean: optee-os-clean-common
+	rm -rf ${OPTEE_OS_PATH}/out/
+
+################################################################################
+# U-Boot
+################################################################################
+
+U-BOOT_EXPORTS = CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+U-BOOT_DEFCONFIG_COMMON_FILES := $(U-BOOT_PATH)/configs/amd_versal2_virt_defconfig \
+			$(CURDIR)/kconfigs/u-boot_versal2.conf
+
+
+u-boot-defconfig: $(U-BOOT_DEFCONFIG_COMMON_FILES)
+	cd $(U-BOOT_PATH) && \
+                ARCH=arm64 \
+                scripts/kconfig/merge_config.sh $(U-BOOT_DEFCONFIG_COMMON_FILES)
+
+u-boot: u-boot-defconfig
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH)
+
+u-boot-defconfig-clean:
+	rm -f $(U-BOOT_PATH)/.config
+
+u-boot-clean: u-boot-defconfig-clean
+	$(U-BOOT_EXPORTS) $(MAKE) -C $(U-BOOT_PATH) clean
+
+###############################################################################
+# Device-Tree
+###############################################################################
+dtbo: u-boot
+	${LINUX_PATH}/scripts/dtc/dtc -@ -I dts -O dtb -o ./versal2/versal2-memory-reservation.dtbo ./versal2/versal2-memory-reservation.dtso
+	@$(foreach dtb,$(wildcard $(U-BOOT_PATH)/arch/arm/dts/versal2-*.dtb), \
+		${LINUX_PATH}/scripts/dtc/fdtoverlay -i $(dtb) -o $(dtb) ./versal2/versal2-memory-reservation.dtbo ; \
+		echo "Applied overlay to $(dtb)";)
+
+dtbo-clean:
+	rm -f ./versal2/versal2-memory-reservation.dtbo
+
+################################################################################
+# Linux kernel
+################################################################################
+
+LINUX_DEFCONFIG_COMMON_ARCH := arm64
+LINUX_DEFCONFIG_COMMON_FILES := \
+		$(LINUX_PATH)/arch/arm64/configs/xilinx_defconfig \
+		$(CURDIR)/kconfigs/versal2.conf
+
+linux-defconfig: $(LINUX_PATH)/.config
+
+LINUX_COMMON_FLAGS += ARCH=arm64 -j8
+
+linux: linux-common
+
+linux-defconfig-clean: linux-defconfig-clean-common
+
+LINUX_CLEAN_COMMON_FLAGS += ARCH=arm64
+
+linux-clean: linux-clean-common
+
+LINUX_CLEANER_COMMON_FLAGS += ARCH=arm64
+
+linux-cleaner: linux-cleaner-common
+
+###############################################################################
+# Buildroot
+###############################################################################
+
+BR2_TARGET_GENERIC_ISSUE="OP-TEE embedded distrib for $(PLATFORM)"
+BR2_TARGET_ROOTFS_EXT2=y
+BR2_PACKAGE_BUSYBOX_WATCHDOG=y
+
+# TF-A, Linux kernel, U-Boot and OP-TEE OS/Client/... are not built from their
+# related Buildroot native package.
+BR2_TARGET_ARM_TRUSTED_FIRMWARE=n
+BR2_LINUX_KERNEL=n
+BR2_TARGET_OPTEE_OS=n
+BR2_TARGET_UBOOT=n
+BR2_PACKAGE_OPTEE_CLIENT=n
+BR2_PACKAGE_OPTEE_TEST=n
+BR2_PACKAGE_OPTEE_EXAMPLES=n
+BR2_PACKAGE_OPTEE_BENCHMARK=n
```

Comment:
Not needed I believe (they should in principle be all be disabled by default). Can you test and confirm?
I can see that this copied from stm32mp1.mk, versal.mk and zynqmp.mk.

Reply:
@jforissier Tested without these in makefile, working as expected!